### PR TITLE
New UI: Staking

### PR DESCRIPTION
**Pull request by kevinunger**
[Initial design](https://xd.adobe.com/view/61961669-fb51-4607-8e31-030fbca6400f-34c5/screen/3b159b14-8f21-4840-943b-7b0d4f186d88/specs/)

Restructuring:
Old structure:
- /stake lists stakeTypes
- Selecting a stakeType will pass the props to Staking, where one can stake/unstake/claim

New structure:
- /stake lists stakeTypes, which now have all the data/methods of Staking

Changes:
- cleanup
- +progressBar 
- +total amount of staked nfts/token in a pool
- +addded buttons to stake/unstaked by ratio and max.
- contract calls are only made if defaultAccount is set. Should now load everything without error 
- available tokens, total staked tokens and staked tokens are rounded (only in template ofc)
- added comments where (accent) colors need to be changed and to which #code when we want to change the theme when the redesign is done completely

ToDo:
- [x] cleanup
- [x] i18n
- [x] TS
- [ ] testing

SSs:
Initial view:
image
Trying to unstake:
image
Staking land:
image
image
 
Mobile:
image

### CODE REVIEW

**Comment by Sellie96** [comment]
image

This message needs changing

File: frontend/src/contracts.ts [hunk 1]
```diff
@@ -98,8 +98,9 @@ export function getConfigValue(key: string): any {
     return process.env[key];
   }
 
-  if(process.env.NODE_ENV === 'development') return '';
+  // if(process.env.NODE_ENV === 'development') return '';
   const env = window.location.href.startsWith('https://test') ? 'test' : 'production';
+  // const env = 'test';
```
Comment (Sellie96) [inline frontend/src/contracts.ts:103]: These need uncommenting

File: frontend/src/components/StakeSelectorItem.vue [hunk 2]
```diff
@@ -82,136 +147,663 @@ export default {
...
-      let exclamationMark = '';
+    currentRewardEarnedFormatted(){
+      return toBN(this.currentRewardEarned).dividedBy(1e18);
+    },
+    totalStakedFormatted(){
+      return this.isNftStaking ? this.totalStaked: toBN(this.totalStaked).dividedBy(1e18).toFixed(0);
+    },
+    walletBalanceFormatted() {
+      return toBN(this.walletBalance).dividedBy(1e18).toFixed(3);
+      // return this.walletBalance;
```
Comment (Sellie96) [inline frontend/src/components/StakeSelectorItem.vue:211]: Remove if not used

File: frontend/src/components/StakeSelectorItem.vue [hunk 3]
```diff
@@ -82,136 +147,663 @@ export default {
...
+    this.stakeUnlockTimeLeftCurrentEstimate = this.unlockTimeLeftInternal;
+    this.stakeRewardDistributionTimeLeftCurrentEstimate = this.rewardDistributionTimeLeftInternal;
+
+    this.stakeRewardProgressInterval = setInterval(async () => {
+      await this.fetchStakeDetails({ stakeType: this.stakeType });
+    }, 10 * 1000);
+
+    this.secondsInterval = setInterval(() => {
+      this.updateEstimates();
+    }, 1000);
```
Comment (Sellie96) [inline frontend/src/components/StakeSelectorItem.vue:182]: Are these something that is required to be updated so often?

File: frontend/src/components/StakeSelectorItem.vue [hunk 4]
```diff
@@ -82,136 +147,663 @@ export default {
...
+
+      if (
+        this.bigNumberAmount.gt(this.remainingCapacityForWithdraw) &&
+        !this.isDeposit
+      ) {
+        return 'notEnoughFundsInExitPool';
+      }
+
+      return 'ok';
+    },
```
Comment (Sellie96) [inline frontend/src/components/StakeSelectorItem.vue:339]: Nasty function, hard to follow. Would appreciate some clean-up splitting it out or some commenting to explain what it is doing for future readers.

Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue:339]: I will. This is a function I just copied from the old staking component.

File: frontend/src/components/StakeSelectorItem.vue (hunk 3 above)
Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue:182]: Not really, but seeing the countdown in the progressbar is good for the user i think. updateEstimates() is only decrementing two vars - I think that's fine.

**Comment by kevinunger** [comment]
@Sellie96 Thanks. Yeah most of the logic / state was already there. I'll try to simplify some, especially the currentState, and add types.

File: frontend/src/views/SelectStakeType.vue [hunk 5]
```diff
@@ -1,6 +1,7 @@
 <template>
   <div class="body main-font">
-    <ul class="stake-select-list" v-if="true">
+    <h1 class="text-center">Staking</h1>
```
Comment (Sellie96) [inline frontend/src/views/SelectStakeType.vue:3]: Missed a translation here

File: frontend/src/components/StakeSelectorItem.vue [hunk 6]
```diff
@@ -1,217 +1,874 @@
...
+  claimKingReward():
+  Promise<void>;
+  stakeUnclaimedRewards(payload: {stakeType: StakeType;}):
+  Promise<void>;
+  claimReward(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<void>;
+  getOwnedLandIdsWithTier():
+  Promise<LandIds[]>;
+  getStakedIds(payload: {stakeType: StakeType | NftStakeType;}):
+  Promise<string[]>;
```
Comment (Chriserus) [inline frontend/src/components/StakeSelectorItem.vue:169]: What is this formatting :O

File: frontend/src/components/StakeSelectorItem.vue [hunk 7]
```diff
@@ -1,217 +1,874 @@
...
+}
+
+enum CurrentState {
+  connectWallet,
```
Comment (Chriserus) [inline frontend/src/components/StakeSelectorItem.vue:173]: Generally enum values should be capitalized, like, connectWallet -> CONNECT_WALLET

File: frontend/src/components/StakeSelectorItem.vue [hunk 8]
```diff
@@ -1,217 +1,874 @@
...
+  waiting,
+  inputIsZero,
+  insufficientBalance,
+  notEnoughFundsInExitPool,
+  notEnoughFundsInExitPoolForFee,
+  ok,
+}
+
+enum RewardClaimState {
+  loading,
```
Comment (Chriserus) [inline frontend/src/components/StakeSelectorItem.vue:186]: Same here, capitalization

File: frontend/src/components/StakeSelectorItem.vue [hunk 9]
```diff
@@ -1,217 +1,874 @@
...
+  unlockTimeLeft: number;
+}
+
+type AllStakeTypes = StakeType | NftStakeType;
+
+export default Vue.extend({
+  components: {
+    Multiselect,
+  },
+  props: {
```
Comment (Chriserus) [inline frontend/src/components/StakeSelectorItem.vue:215]: If there are required props, please mark them as required, for future use

File: frontend/src/components/StakeSelectorItem.vue [hunk 10]
```diff
@@ -1,217 +1,874 @@
...
+    },
+    totalStaked: {
+      type: String,
+    },
+    walletBalance: {
+      type: String,
+    },
+  },
+  data() {
+    return {
```
Comment (Chriserus) [inline frontend/src/components/StakeSelectorItem.vue:254]: It's fine, but in general we add "Data" interface and hold types there, not directly here. The interface always keeps tabs on what's missing and the IDE will inform you if the type is incorrect.

File: frontend/src/components/StakeSelectorItem.vue [hunk 11]
```diff
@@ -1,217 +1,874 @@
...
-};
+});
 </script>
 
 <style scoped>
 .container {
-  background: rgb(22, 22, 22);
-  background: linear-gradient(180deg, rgba(22, 22, 22, 1) 0%, rgba(16, 17, 17, 1) 100%);
-  padding: 1rem;
+  background: rgb(22, 22, 22); /* change to: background: #000E1D; */
```
Comment (Chriserus) [inline frontend/src/components/StakeSelectorItem.vue:653]: A TODO I suppose?

Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue:653]: Yes. When the whole new game ui is done we want to change the color theme.

File: frontend/src/views/SelectStakeType.vue (hunk 5 above)
Comment (kevinunger) [reply frontend/src/views/SelectStakeType.vue]: resolved with [60bc5f4](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/60bc5f4ef96751735b4153be50b973723c6cb7a9)

File: frontend/src/components/StakeSelectorItem.vue (hunk 7 above)
Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue]: done [9115fbe](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/9115fbe2306749216362df1ebece5b1d75a76b2b)

File: frontend/src/components/StakeSelectorItem.vue (hunk 8 above)
Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue]: done [9115fbe](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/9115fbe2306749216362df1ebece5b1d75a76b2b)

File: frontend/src/components/StakeSelectorItem.vue (hunk 9 above)
Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue:221]: done [6839728](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/6839728c34bb22d46c9a4328c2f8e7680f731db3)

File: frontend/src/components/StakeSelectorItem.vue (hunk 10 above)
Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue:271]: [394edfb](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/394edfb4636310a2c04bc4448a64b27060a3d1c1)

File: frontend/src/components/StakeSelectorItem.vue (hunk 6 above)
Comment (kevinunger) [reply frontend/src/components/StakeSelectorItem.vue]: Done:
[77ebd5e](https://github.com/CryptoBlades/cryptoblades/pull/1287/commits/77ebd5e03dde77fd7b81345a1d03ccda2bfb96ee)

**Review by Sellie96 — APPROVED** [review]
